jgrapht-core/src/main/java/org/jgrapht/graph/AbstractBaseGraph.java

```diff
@@ -516,55 +486,24 @@ public void setEdgeWeight(E e, double weight)
         ((DefaultWeightedEdge) e).weight = weight;
     }
 
-    private Specifics<V, E> createSpecifics()
+    /**
+     * Create the specifics for this graph. Subclasses can override this method in order to adjust
+     * the specifics and thus the space-time tradeoffs of the graph implementation.
+     * 
+     * @return the specifics used by this graph
+     */
+    protected Specifics<V, E> createSpecifics()
```

Comment:
We could do this also in 2 releases. I hoped to get away with this one as it is somehow on the "backend" of the library.